pytorch_lightning/trainer/training_loop.py

```diff
@@ -232,8 +232,14 @@ def on_train_epoch_start(self, epoch):
         model = self.trainer.get_model()
 
         # reset train dataloader
-        if self.trainer.reload_dataloaders_every_epoch:
+        assert self.trainer.reload_dataloaders_every_n_epochs >= 0
+        reload_dl_every_n_epochs = self.trainer.reload_dataloaders_every_n_epochs
+
+        if reload_dl_every_n_epochs and (not epoch % reload_dl_every_n_epochs):
+            self.trainer.should_reload_dl_epoch = True
```

Comment:
let's use the property here. the condition should reside in the property that will tell whether to reload the dataloaders or not.